addons/sale_timesheet/models/project.py

```diff
@@ -205,6 +205,7 @@ def default_get(self, fields):
     analytic_account_id = fields.Many2one('account.analytic.account', related='sale_order_id.analytic_account_id')
     bill_type = fields.Selection(related="project_id.bill_type")
     pricing_type = fields.Selection(related="project_id.pricing_type")
+    sale_line_id = fields.Many2one(domain="[('company_id', '=', company_id), ('is_service', '=', True), ('order_partner_id', 'child_of', commercial_partner_id), ('is_expense', '=', False), ('state', 'in', ['sale', 'done']), '|', ('project_id.pricing_type', 'not in', ('project_rate', 'employee_rate')), ('order_id', '=?', project_sale_order_id)]")
```

Comment:
Although I know I told you that we needed an exception on the project_sale_order_id check when pricing_type was either project_rate or employee_rate, I think writing it that way will be easier to understand:
```suggestion
    sale_line_id = fields.Many2one(domain="[('company_id', '=', company_id), ('is_service', '=', True), ('order_partner_id', 'child_of', commercial_partner_id), ('is_expense', '=', False), ('state', 'in', ['sale', 'done']), '|', ('project_id.pricing_type', '=', 'task_rate'), ('order_id', '=?', project_sale_order_id)]")
```
What do you think?